### PR TITLE
Add pay to mint feature to nft minting

### PR DESCRIPTION
**Pull request by hochiw**
- Added genesis param 'MintPriceDenom'
- Added the ability to mint nft as a non-owner when nft class mint price is larger than zero
- Retain owner checking mechanics when mint price is zero such that only the owner of the class is able to mint nfts
- Added test cases for when mint price is larger than zero and minter has sufficient funds vs insufficient funds

- Manually tested with a single node locally
    - Class with mint price = 0, minter is owner, can mint
    - Class with mint price = 0, minter is not owner, cannot mint
    - Class with mint price = 5000000000, minter is owner, can mint
    - Class with mint price = 5000000000, minter is not owner, minter has insufficient funds, cannot mint
    - Class with mint price = 5000000000, minter is not owner, minter has sufficient funds, can mint

Note: it seems that the mint price doesn't appear in the transaction message when minting an nft as a non-owner, it's expected based on how we coded it but im not sure if it is a necessary information to show since iirc stargaze doesn't do it as well.

based on oursky/likecoin-chain#171 
refs oursky/likecoin-chain#150

### CODE REVIEW

**Comment by chihimng** [comment]
@hochiw 
I think we should decouple whether the class for public pay-to-mint & the price, so we can allow the use case where a class is for free claiming

**Comment by chihimng** [comment]
sorry should have discussed more before implementation 🙇

File: x/likenft/keeper/msg_server_mint_nft_test.go [hunk 1]
```diff
@@ -744,7 +802,7 @@ func TestMintNFTInvalidUserAddress(t *testing.T) {
 	ctrl.Finish()
 }
 
-func TestMintNFTUserNotOwner(t *testing.T) {
+func TestMintPayToMintNFTNormal(t *testing.T) {
```
Comment (chihimng) [inline x/likenft/keeper/msg_server_mint_nft_test.go:805]: Did we cover disabled pay to mint?

**Review by chihimng — CHANGES_REQUESTED** [review]
Also we might need to update some terminology after latest plan at #192

**Comment by hochiw** [comment]
@chihimng updated, also refactored a little bit to prevent calling `GetContentIdRecord` twice.

**Review by chihimng — APPROVED** [review]

**Pull request merged**